### PR TITLE
Deploy UDC during service deploy

### LINKED ISSUE #499: Incorrect argument count in deployment script
When i try to deploy the service contracts, I see a wrong number of arguments somewhere.
```
(venv) ~/src/raiden-contracts(master) $ python -m raiden_contracts.deploy services \
> --rpc-provider <snip> \
> --private-key /home/yh/.local/share/io.parity.ethereum/keys/kovan/UTC--2019-01-08T17-09-24Z--fcf0e5e3-7d74-7abc-773e-a2ccab085547 \
> --gas-price 30 --gas-limit 6000000 \
> --token-address 0xd0a1e359811322d97991e03f863a0c30c2cf029c
Web3 provider is RPC connection <snip>
Enter the private key password:
DEBUG:__main__:Deploying ServiceRegistry txHash=0xd9fda14d3f051ad079a826f70fc8f1fab2c294fcbc126388082906ec2981861a, contracts version 0.5.0
INFO:__main__:ServiceRegistry address: 0xb7E1e47b47f7A6868544EAD4D205F913d9d4D9e8. Gas used: 405912
Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/yh/src/raiden-contracts/raiden_contracts/deploy/__main__.py", line 831, in <module>
    main()
  File "/home/yh/src/raiden-contracts/venv/lib/python3.6/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/yh/src/raiden-contracts/venv/lib/python3.6/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/yh/src/raiden-contracts/venv/lib/python3.6/site-packages/click/core.py", line 1163, in invoke
    rv.append(sub_ctx.command.invoke(sub_ctx))
  File "/home/yh/src/raiden-contracts/venv/lib/python3.6/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/yh/src/raiden-contracts/venv/lib/python3.6/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/yh/src/raiden-contracts/raiden_contracts/deploy/__main__.py", line 177, in wrapper
    return func(*args, **kwargs)
  File "/home/yh/src/raiden-contracts/venv/lib/python3.6/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/yh/src/raiden-contracts/raiden_contracts/deploy/__main__.py", line 295, in services
    deployed_contracts_info = deploy_service_contracts(deployer, token_address)
  File "/home/yh/src/raiden-contracts/raiden_contracts/deploy/__main__.py", line 521, in deploy_service_contracts
    monitoring_service_constructor_args,
  File "/home/yh/src/raiden-contracts/raiden_contracts/deploy/__main__.py", line 108, in deploy
    txhash = self.send_deployment_transaction(contract, args)
  File "/home/yh/src/raiden-contracts/raiden_contracts/deploy/__main__.py", line 132, in send_deployment_transaction
    txhash = contract.constructor(*args).transact(
  File "/home/yh/src/raiden-contracts/venv/lib/python3.6/site-packages/web3/contract.py", line 342, in constructor
    **kwargs)
  File "/home/yh/src/raiden-contracts/venv/lib/python3.6/site-packages/web3/contract.py", line 798, in __init__
    self.data_in_transaction = self._encode_data_in_transaction(*args, **kwargs)
  File "/home/yh/src/raiden-contracts/venv/lib/python3.6/site-packages/web3/utils/decorators.py", line 14, in _wrapper
    return self.method(obj, *args, **kwargs)
  File "/home/yh/src/raiden-contracts/venv/lib/python3.6/site-packages/web3/contract.py", line 810, in _encode_data_in_transaction
    arguments = merge_args_and_kwargs(constructor_abi, args, kwargs)
  File "/home/yh/src/raiden-contracts/venv/lib/python3.6/site-packages/web3/utils/abi.py", line 182, in merge_args_and_kwargs
    len(args) + len(kwargs),
TypeError: Incorrect argument count.  Expected '3'.  Got '2'
```

I found this issue during #481 .

### PR DESCRIPTION
**Pull request by pirapira**
After this PR, `python -m raiden_contracts.deploy services` deploys a `UserDeposit` contract.

Fixes #499.

### CODE REVIEW

**Comment by pirapira** [comment]
One possible refactoring I noticed (repetitive translation of transaction receipts into deployment data): https://github.com/raiden-network/raiden-contracts/issues/501 .

**Comment by pirapira** [comment]
`verify_deployed_service_contracts()` is also repetitive. https://github.com/raiden-network/raiden-contracts/issues/502

File: raiden_contracts/deploy/__main__.py [hunk 1]
```diff
@@ -729,6 +746,28 @@ def verify_deployed_service_contracts(
...
+
+    # We need to also check the constructor parameters against the chain
+    constructor_arguments = deployment_data['contracts'][
+        CONTRACT_USER_DEPOSIT
+    ]['constructor_arguments']
+    assert len(constructor_arguments) == 1
+    assert to_checksum_address(
+        user_deposit.functions.token().call(),
+    ) == token_address
+    assert token_address == constructor_arguments[0]
```
Comment (karlb) [inline raiden_contracts/deploy/__main__.py]: Isn't it somewhat redundant to do both
```python
assert to_checksum_address(
    user_deposit.functions.token().call(),
) == token_address
```
and
```python
constructor_arguments = deployment_data['contracts'][
    CONTRACT_USER_DEPOSIT
]['constructor_arguments']
assert len(constructor_arguments) == 1
assert token_address == constructor_arguments[0]
```

**Comment by pirapira** [comment]
> The UDC also needs a call to its init function. Is this left out because the OneToN contract has no deployment code, yet?

Yes, it's half-baked. Today I'm focusing on putting these onchain: https://github.com/raiden-network/raiden-contracts/issues/481

Calling `init` is planned for tomorrow: https://github.com/raiden-network/raiden-contracts/issues/482

File: raiden_contracts/deploy/__main__.py (hunk 1 above)
Comment (pirapira) [reply raiden_contracts/deploy/__main__.py]: The first checks what's onchain.  The second checks what we remember from the deployment time.

So the second check might be redundant, but I'm not sure. `deployment_data` sometimes comes from the filesystem.

**Pull request merged**